usual/tls/tls.c

```diff
@@ -742,4 +742,58 @@ tls_close(struct tls *ctx)
 	return (rv);
 }
 
+static bool
+tls_config_keypair_equal(struct tls_keypair *left_tls_keypair, struct tls_keypair *right_tls_keypair)
+{
+	return (strings_equal(left_tls_keypair->cert_file, right_tls_keypair->cert_file) &&
+		memcmp(left_tls_keypair->cert_mem, right_tls_keypair->cert_mem, left_tls_keypair->cert_len) == 0 &&
+		strings_equal(left_tls_keypair->key_file, right_tls_keypair->key_file) &&
+		memcmp(left_tls_keypair->key_mem, right_tls_keypair->key_mem, left_tls_keypair->key_len) == 0);
+}
+
```

Comment:
`tls_config_keypairs_equal` is pretty complicated and I hope there is a better way to do it. Either way, though, I'd just as soon delete it. As I noted on the pgbouncer PR, I verified that these objects are always NULL in the place where we compare TLS configs: https://github.com/pgbouncer/pgbouncer/pull/1157#discussion_r1744040807

Reply:
I agree it's pretty complicated. According to my reading of the code, the keypairs shouldn't be always NULL in pgbouncer. Because `tls_config_set_key_file` calls `tls_keypair_set_key_file`, which means the keypair field should contain the key. So it seems important to keep these.

I cannot find a place where the `next` field is actually used though.